jobs/postgres/templates/bin/functions

```diff
@@ -10,9 +10,27 @@ is_master() {
   else
     opts="$opts -p <%= p('postgres.config')["port"] || 6432 %>"
   fi
-  tf=$(echo $(psql $opts postgres -t -c 'SELECT pg_is_in_recovery()' 2>/dev/null));
-  if [[ "$tf" == "f" ]]; then
-    return 0
-  fi
-  return 1
-}
+
+  # psql can experience transient issues (like connection reset)
+  # make is_master more resilient against these kinds of errors
+  declare -i error_count=0
+  while [[ $error_count < 3 ]] ; do
```

Comment:
Look into bash's math mode `(( $error_count < 3 ))`; i _think_ it's more correct.  It also may allow you to get rid of the explicit `declare -i`, since it will always treat "0" as the numeric quantity zero.